x-pack/legacy/plugins/siem/public/pages/detection_engine/create_rule/components/description_step/index.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiBadge, EuiDescriptionList, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { isEmpty, chunk, get, pick } from 'lodash/fp';
+import React, { memo, ReactNode } from 'react';
+
+import {
+  IIndexPattern,
+  esFilters,
+  Query,
+  utils,
+  FilterLabel,
+} from '../../../../../../../../../../src/plugins/data/public';
+
+import { FormSchema } from '../shared_imports';
+import * as I18n from './translations';
+
+interface StepRuleDescriptionProps {
+  data: unknown;
+  indexPatterns?: IIndexPattern;
+  schema: FormSchema;
+}
+
+export const StepRuleDescription = memo<StepRuleDescriptionProps>(
+  ({ data, indexPatterns, schema }) => {
+    const keys = Object.keys(schema);
+    return (
+      <EuiFlexGroup gutterSize="none" direction="row" justifyContent="spaceAround">
+        {chunk(keys.includes('queryBar') ? 3 : Math.ceil(keys.length / 2), keys).map(key => (
+          <EuiFlexItem grow={false}>
+            <EuiDescriptionList
+              listItems={buildListItems(data, pick(key, schema), indexPatterns)}
+            />
+          </EuiFlexItem>
+        ))}
+      </EuiFlexGroup>
+    );
+  }
+);
+
+interface ListItems {
+  title: NonNullable<ReactNode>;
+  description: NonNullable<ReactNode>;
+}
+
+const buildListItems = (
+  data: unknown,
+  schema: FormSchema,
+  indexPatterns?: IIndexPattern
+): ListItems[] =>
+  Object.keys(schema).reduce<ListItems[]>(
+    (acc, field) => [
+      ...acc,
+      ...getDescriptionItem(field, get([field, 'label'], schema), data, indexPatterns),
+    ],
+    []
+  );
+
+const getDescriptionItem = (
+  field: string,
+  label: string,
+  value: unknown,
+  indexPatterns?: IIndexPattern
+): ListItems[] => {
+  if (field === 'queryBar' && indexPatterns != null) {
+    const filters = get('queryBar.filters', value) as esFilters.Filter[];
+    const query = get('queryBar.query', value) as Query;
+    const savedId = get('queryBar.saved_id', value);
+    let items: ListItems[] = [];
+    if (!isEmpty(filters)) {
+      items = [
+        ...items,
+        {
+          title: <>{I18n.FILTERS_LABEL}</>,
+          description: (
+            <EuiFlexGroup wrap responsive={false} gutterSize="xs">
```

Comment:
Not sure how much can be done here, but large `filter`'s will squish the left content pane even when there is extra room for the `filters`'s section to expand. 

e.g.

![image](https://user-images.githubusercontent.com/2946766/69585903-395f0780-0f9e-11ea-9f5a-b6a0d16ae00a.png)

Here it is with a larger query for ref ( which looks great 🎉 ):
![image](https://user-images.githubusercontent.com/2946766/69586345-8abbc680-0f9f-11ea-9d29-e5f406d62954.png)

Reply:
Similarly -- do we want to do any sort of truncation within this preview view, or is it beneficial to have the entire value?

I.e. how big can descriptions get and is this useful?

![image](https://user-images.githubusercontent.com/2946766/69586679-8d6aeb80-0fa0-11ea-8afb-e611d13a24ce.png)